### PR TITLE
Drake/feature/i2c interface

### PR DESCRIPTION
**Pull request by DrakBoul**
This interface is pretty bare bones at the moment, and essentially is just an extremely thin wrapper for the library I use since it offers no extra functionality. I would like to hear back which feature would be useful to include in the i2c interface besides basic send and read functions. Please leave any comments on what I should add/improve.

### CODE REVIEW

File: ex3_shared_libs/interfaces/i2c/src/lib.rs [hunk 1]
```diff
@@ -5,16 +5,15 @@ use i2cdev::linux::{LinuxI2CDevice, LinuxI2CError};
 pub trait Interface {
     // send data as bytes to the interface as a shared byte type slice.
     // Returns the number of bytes sent
-    fn send(&mut self, data: &[u8]) -> Result<usize, LinuxI2CError>;
+    fn send(&mut self, data: &[u8]) -> Result<(), LinuxI2CError>;
 
     // Read byte data from the interfaace into a byte slice buffer.
     // Returns number of bytes read
-    fn read(&mut self, buffer: &mut [u8]) -> Result<usize, LinuxI2CError>;
+    fn read(&mut self, buffer: &mut [u8]) -> Result<(), LinuxI2CError>;
```
Comment (rcunrau) [inline ex3_shared_libs/interfaces/i2c/src/lib.rs:12]: The comment still says send/read return number of bytes sent. Kind of odd that they don't?

**Review by rcunrau — APPROVED** [review]
Thank you!

File: Cargo.toml [hunk 2]
```diff
@@ -15,8 +15,9 @@ members = [
     "ex3_obc_fsw/handlers/shell_handler",
     "ex3_obc_fsw/scheduler",
     "ex3_shared_libs/bulk_msg_slicing",
-    "ex3_shared_libs/common",
+    "ex3_shared_libs/common", "ex3_shared_libs/interfaces/i2c",
     "ex3_shared_libs/interfaces/ipc",
+    "ex3_shared_libs/interfaces/i2c",
```
[the remark this answers is not in the document]
Comment (DrakBoul) [reply Cargo.toml:20]: It kind of looks that way but these are actually two different directories. The first one is for Interprocess communication and the second is the I2C interface (IPC vs. I2C).

File: ex3_shared_libs/interfaces/i2c/src/lib.rs (hunk 1 above)
Comment (DrakBoul) [reply ex3_shared_libs/interfaces/i2c/src/lib.rs]: Fixed.

**Pull request merged**